pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -8,15 +8,19 @@ import (
 )
 
 type AccessMode string
+type DHCPConfigMode string
 
 const (
-	AccessModePublic  string = "Public"
-	AccessModePrivate string = "Private"
-	AccessModeProject string = "PrivateTGW"
+	AccessModePublic          string = "Public"
+	AccessModePrivate         string = "Private"
+	AccessModeProject         string = "PrivateTGW"
+	DHCPConfigModeDeactivated string = "DHCPDeactivated"
+	DHCPConfigModeServer      string = "DHCPServer"
+	DHCPConfigModeRelay       string = "DHCPRelay"
 )
 
 // SubnetSpec defines the desired state of Subnet.
-// +kubebuilder:validation:XValidation:rule="!has(oldSelf.DHCPConfig) || has(self.DHCPConfig)", message="DHCPConfig is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.subnetDHCPConfig) || has(self.subnetDHCPConfig)", message="subnetDHCPConfig is required once set"
```

Comment:
if subnetDHCPConfig.mode is DHCPDeactivated, then we do not support it to be updated to other mode, do we plan to do that in webhook? Can it be supported via kubebuilder XValidation?

Reply:
It could be done with XValidation, updated the validation rule, thanks!